test/functional/update_v2/update-iterative-and-full.bats

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -L -n test-bundle-1 -f /foo/test-file1 "$TEST_NAME"
+	create_bundle -L -n test-bundle-2 -f /foo/test-file2 "$TEST_NAME"
+	create_version "$TEST_NAME" 20 10
+	update_bundle -i "$TEST_NAME" test-bundle-1 --update /foo/test-file1
+	create_version "$TEST_NAME" 30 20
+	update_bundle -i "$TEST_NAME" test-bundle-1 --update /foo/test-file1
+	update_bundle -i "$TEST_NAME" test-bundle-2 --update /foo/test-file2
+
+}
+
+@test "update works correctly using both full and iterative manifests in the same operation" {
+
+	run sudo sh -c "$SWUPD update $SWUPD_OPTS_KEEPCACHE"
+
+	assert_status_is 0
+
+	# verify target files
+	assert_file_exists "$TARGETDIR"/foo/test-file1
```

Comment:
This verifies that the files exist at the correct locations, but it does not validate their contents for a successful update. The hashes of the files could be compared against the manifest to better validate a successful update. You could use something like the following:

hash1=$("$SWUPD" hashdump "TARGETDIR"/foo/test-file1)
hash2=$(get_hash_from_manifest "$WEBDIR"/30/test-bundle-1 /foo/test-file1)
assert_equal "$hash1" "$hash2"

Reply:
Well, there are a few reasons why I didn't think on including that on the tests initially:
1) the most important one is that the actual process of applying the updates (deltas) on the files was not changed, what changed was the way the manifests were handled, so there is very small risk that the files would update incorrectly IMO. 
2) The second point is related with the first one, update and verify tests are slow tests due to all the set up they need to run, they take an average of 14 seconds each one, so I was concerned about adding extra validations that could add up more time.
3) Finally one problem I see with checking the file content like that is that we are using the system under test (swupd) to validate itself, which is not a good practice because we could mask bugs. Ideally we would have to write an independent function to validate these hashes in the testlib, or use a different tool, but I know this is probably not feasible.

Reply:
I see. This is fine with me.